SampleARApp/SampleARApp/Utils/CoordinatesAdapter.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  CoordinatesAdapter.swift
+//  SampleARApp
+//
+//  Created by Tony Ngo on 11.10.2022.
+//
+
+import UIKit
+
+struct CoordinatesAdapter {
+    var imageSize: CGSize = .zero
+    let screenSize: CGSize
+}
+
+extension CoordinatesAdapter {
+    func screenCoordinates(normalizedImagePoint: CGPoint) -> CGPoint {
+        let normalizedScreenPoint = normalizeScreen(point: normalizedImagePoint)
+        return CGPoint(
+            x: screenSize.width * normalizedScreenPoint.x,
+            y: screenSize.height * normalizedScreenPoint.y
+        )
+    }
+}
+
+private extension CoordinatesAdapter {
+    /// A height scale factor between the color image and the screen.
+    var multiplier: Double {
+        imageSize.height / screenSize.height
+    }
+
+    /// A scaled screen size by `multiplier`.
+    var multipliedScreenSize: CGSize {
+        CGSize(
+            width: screenSize.width * multiplier,
+            height: screenSize.height * multiplier
+        )
+    }
+
+    /// A normalized width value of non-visible area relative to the color image.
+    var nonVisibleAreaWidth: Double {
+        (imageSize.width - multipliedScreenSize.width) / 2 / imageSize.width
+    }
+
+    /// A normalized range representing a visible area relative to the color image.
+    var visibleAreaRange: ClosedRange<Double> {
+        nonVisibleAreaWidth...1 - nonVisibleAreaWidth
+    }
+
+    /// A normalized width value of visible area relative to the color image.
+    var visibleAreaWidth: Double {
+        visibleAreaRange.upperBound - visibleAreaRange.lowerBound
+    }
+
+    /// Returns a normalized `CGPoint` representing a point in the screen.
+    func normalizeScreen(point: CGPoint) -> CGPoint {
+        CGPoint(
+            x: (point.x - visibleAreaRange.lowerBound) / visibleAreaWidth,
+            y: point.y
+        )
+    }
+}
```

Comment:
> note here to files below?

The description is quite long so I'd rather include it in a README file than here. I'll add it.

> do we use somewhere ship.scn? and texture.png?
> can we add somewhere (live example, or presentation) how to create scn files?

Not now, but I left it there so that we can mention that such a thing exists. I'll add a new example, showing the creation of a custom scene.